plugins/rsync_server.py

```diff
@@ -0,0 +1,157 @@
+                 include=None,
+                 time_key='rsync_time',
+                 username='pi',
+                 **output_step_kwargs):
+        """
+        Sync files to a server via rsync.
+
+        This class will "inject" a new entry into the dictionary Brokkr passes
+        in pipelines that tracks the last time the rsync was performed.
+
+        If you want to rsync two different paths in the same pipeline, you
+        would provide two different steps and unique `time_key`s for each step.
+
+        Parameters
+        ----------
+        server : str
+            The server to sync the files to.
+        local_path : str
+            The local path to the files to be synced.
+        server_path : str
+            The path on the server the files will be synced to
+        username : str
+            The user name on the server.
+        update_time : int
+            The number of seconds that must elapse between successful syncs.
+            Sets the cadence of syncing to the server.
+        include : str
+            Typically, a pattern of files to be synced. Passed to the `include` option
+            of rsync.
+        time_key : str
+            The key corresponding to the time to track the last update.
+        output_step_kwargs : **kwargs, optional
+            Keyword arguments to pass to the OutputStep constructor.
+
+        """
+        # Pass arguments to superclass init
+        super().__init__(**output_step_kwargs)
+
+        self._previous_data = None
+
+        self.server = server
+        self.local_path = local_path
+        self.server_path = server_path
+        self.username = username
+        self.update_time = update_time
+        self.include = include
+        self.time_key = time_key
+
+    def execute(self, input_data=None):
+        """
+        Executing an action upon detection an arbitrary condition in the data.
+
+        Parameters
+        ----------
+        input_data : any, optional
+            Per iteration input data passed to this function from previous
+            PipelineSteps. Not used here but retained for compatibility with
+            the generalized PipelineStep API. The default is None.
+
+        Returns
+        -------
+        input_data : same as input_data
+            Input data passed through, for further steps to consume. Note
+            a new key will be added. See documentation for initialization
+            of the class.
+        """
+
+        # Handle first iteration
+        if self._previous_data is None:
+            input_data[self.time_key] = input_data['time']
+            self._previous_data = input_data
+
+        try:
+            # How long since we last updated?
+            dt = input_data['time'].value-self._previous_data[self.time_key].value
+
+            if dt.total_seconds() > self.update_time:
+                # Now, sync the files to the server
+                out = self.rsync()
+                self.logger.info(out)
```

Comment:
```suggestion
                self.logger.info("Rsync output: %s", out)
```

Be a little more clear here and avoid (accidental or malicious) Log4Shell-style injection

Also, its ultimately up to you of course, but you might want to consider using `debug` instead here (and in State Notifier)—in general, the other steps all reserve `info` for things that don't happen every iteration, which can get very noisy very quick and fill up the logs, as opposed to error details or other relatively unusual, extraordinary or once-per-Brokkr-startup/shutdown messages.

```suggestion
                self.logger.debug("Rsync output: %s", out)
```